lib/enjoi.js

```diff
@@ -156,6 +156,14 @@ module.exports = function enjoi(schema, options) {
     function object(current) {
         var joischema = Joi.object(resolveproperties(current));
 
+        if (Thing.isObject(current.additionalProperties)) {
+          joischema = joischema.keys(resolveproperties({properties: current.additionalProperties}));
+        }
+
+        if (current.additionalProperties) {
```

Comment:
> The value of "additionalProperties" MUST be a boolean or an object. If it is an object, it MUST also be a valid JSON Schema.

Probably need to do some further variation around the existence of this property. If it is an `object`, then is `unknown(true)` too broad? Perhaps `alternatives` needs to be determined at that point?